x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/session/Configuration.java

```diff
@@ -18,7 +20,8 @@
 
     public Configuration(ZoneId zi, String username, String clusterName) {
         this.zoneId = zi.normalized();
-        this.now = ZonedDateTime.now(zoneId);
+        Clock clock = Clock.system(zoneId);
+        this.now = ZonedDateTime.now(Clock.tick(clock, Duration.ofNanos(1)));
```

Comment:
Thought it would be a nice improvement, since now we can return > millis precision, to be able to return microseconds for the CURRENT_TIMESTAMP/CURRENT_TIME and their aliases, if requested with the precision param. (Default value remains 3 -> millis).